browser_patches/firefox-beta/juggler/protocol/PageHandler.js

```diff
@@ -302,8 +302,59 @@ class PageHandler {
     return await this._contentPage.send('adoptNode', options);
   }
 
-  async ['Page.screenshot'](options) {
-    return await this._contentPage.send('screenshot', options);
+  async ['Page.screenshot']({ mimeType, clip, omitDeviceScaleFactor }) {
+    let lastError = null;
+    for (let i = 0; i < 3; ++i) {
+      try {
+        // This might occasionally throw if called during navigation.
+        // So attempt screenshot at least 3 times.
+        return await this._screenshotInternal({ mimeType, clip, omitDeviceScaleFactor });
+      } catch (error) {
+        lastError = error;
+      }
+    }
+    throw lastError;
+  }
+
+  async _screenshotInternal({ mimeType, clip, omitDeviceScaleFactor }) {
+    const windowInfo = await this._contentPage.send('getWindowInfo');
+    const rect = clip ?
```

Comment:
From our client code, we always pass `clip` here:
https://github.com/microsoft/playwright/blob/main/packages/playwright-core/src/server/firefox/ffPage.ts#L441

We also have the following TODOs, let's fix them right away?
```js
    // TODO: remove fullPage option from Page.screenshot.
    // TODO: remove Page.getBoundingBox method.
```